rpmio/rpmio.c

```diff
@@ -1116,6 +1116,28 @@ int Fseek(FD_t fd, off_t offset, int whence)
     return rc;
 }
 
+int Fsync(FD_t fd)
+{
+    int rc = 0, ec = 0;
+
+    if (fd == NULL)
+	return -1;
+
+    fd = fdLink(fd);
+    for (FDSTACK_t fps = fd->fps; fps != NULL; fps = fdPop(fd)) {
```

Comment:
Mm, AFAICS this ends up destroying the fd. Which doesn't perhaps matter for this particular use, but for a general purpose public API that's not sensical behavior. Just walk the fps->prev chain instead.